clean/jefferson_so_pprr.py

```diff
@@ -0,0 +1,158 @@
+import sys
+sys.path.append('../')
+from lib.path import data_file_path
+import pandas as pd
+from lib.columns import clean_column_names, set_values
+from lib.clean import standardize_desc_cols
+from lib.uid import gen_uid
+
+
+def extract_department_desc(df):
+    departments = df.rank_desc\
+        .str.extract(r'( ?i\.?t\.? ?| ?accounting ?| ?payroll ?| ?purchasing ?| ?property ?| ?narcotics ?| ?baliff ?|'
+                     r' ?intake ?| ?jail ?| ?communications ?| ?propert ?| ?cib ?| ?juvenile ?|'
+                     r' ?internal affairs ?| ?burglary ?| ?central records ?| ?crime lab ?| ?ems ?| ?homicide ?|'
+                     r' ?robbery ?| ?court ?| ?human resources ?| ?accounting ?|'
+                     r' ?academy ?| ?mail ?| ?database ?| ?network ?| ?\, div\. [abc] ?|'
+                     r'\, hear 1|arson ?| ?auto theft ?| ?pawn shop ?| ?management ?|'
+                     r' ?task force ?| ?ncic ?| ?pawn shop ?| ?jttf ?| ?internal management ?|'
+                     r' ?dna lab ?| ?crime scene ?| ?dispatche?r? ?| ?training ?| ?firearms ?|'
+                     r' ?traffic ?| ?economic crime ?| ?shoot squad ?| ?s\.i\.u ?|'
+                     r' ?booking ?| ?jpcc ?| ?transportation ?|'
+                     r'\, ?(.+)?| ?dea ?(.+)?| ?narcoc?e?c?s ?| ?strategic engagement ?|'
+                     r' ?cic ?| ?intelligence ?| ?intensive probation ?|'
+                     r' ?personal violence ?| ?crimes ?| ?missing persons ?|'
+                     r' ?operations ?| ?sales tax ?| ?health insurance ?|'
+                     r' ?grants ?| ?budget ?| ?finance?i?a?l? ?| ?tax (collector|supervisor) ?|'
+                     r' ?public assisgments ?| ?youth programs ?| ?community relations ?|'
+                     r' ?lab services ?|\;? ?technical services| ?prisoner ?|'
+                     r' ?asset forfeure ?| ?strategic engagement ?| ?dlr ?|'
+                     r' ?tech services ?| ?sib ?| ?p\.o\.c\. ?| ?school offense ?|'
+                     r' ?accounts payable ?| ?range ?| ?light shop ?| ?public assignments ?|'
+                     r' ?help desk tech ?| ?print shop ?| ?mang services ?|'
+                     r' ?terminal agency ?| ?handgun perms ?| ?insurance ?| ?vice ?|'
+                     r' ?photo lab ?| ?fingerprint ?| ?polygraph ?| ?evidence ?|'
+                     r' ?digital forensic ?| ?dna ?| ?patrol ?)')
```

Comment:
Be careful of very long regex. It tends to be slow. A lookup table will almost certainly run faster in this case. It will also minimize the corpus so that whatever additional regex that you have to use will also be simpler.

Reply:
Thanks.